src/query/storages/fuse/src/operations/common/processors/sink_commit.rs

```diff
@@ -342,13 +349,22 @@ where F: SnapshotGenerator + Send + 'static
                 .await
                 {
                     Ok(_) => {
-                        if self.transient {
-                            // Removes historical data, if table is transient
+                        if Self::is_truncate() {
+                            catalog
+                                .truncate_table(&table_info, TruncateTableReq {
```

Comment:
Maybe `catalog.truncate_table()` should not be called when in the delete fast path.

Reply:
> Maybe `catalog.truncate_table()` should not be called when in the delete fast path. Delete stmt should only delete the table data, not influence the `copied_files` in meta.

Thx, fixed